pkg/cephmgr/file/agent.go

```diff
@@ -0,0 +1,206 @@
+func (a *mdsAgent) Initialize(context *clusterd.Context) error {
+	return nil
+}
+
+// configure an MDS on the local node if it is in the desired state
+func (a *mdsAgent) ConfigureLocalService(context *clusterd.Context) error {
+	mdsID, err := getMDSID(context.EtcdClient, context.NodeID)
+	if err != nil {
+		if util.IsEtcdKeyNotFound(err) {
+			// the mds is not in the desired state on this node, so ensure it is not running
+			return a.DestroyLocalService(context)
+		}
+
+		return fmt.Errorf("failed to load mds id. %+v", err)
+	}
+
+	cluster, err := mon.LoadClusterInfo(context.EtcdClient)
+	if err != nil {
+		return fmt.Errorf("failed to load cluster info. %+v", err)
+	}
+
+	_, err = mon.GenerateConnectionConfigFile(context, cluster, getMDSConfDir(context.ConfigDir, mdsID),
+		fmt.Sprintf("mds.%s", mdsID), getMDSKeyringPath(context.ConfigDir, mdsID))
+	if err != nil {
+		return fmt.Errorf("failed to create mds config file. %+v", err)
+	}
+
+	// Connect to the ceph cluster
+	conn, err := mon.ConnectToClusterAsAdmin(context, a.factory, cluster)
+	if err != nil {
+		return fmt.Errorf("failed to connect to cluster. %+v", err)
+	}
+	defer conn.Shutdown()
+
+	err = createMDSKeyring(conn, mdsID, context.ConfigDir)
+	if err != nil {
+		return fmt.Errorf("failed to create mds keyring. %+v", err)
+	}
+
+	err = a.startMDS(context, cluster, mdsID)
+	if err != nil {
+		return fmt.Errorf("failed to start mds process. %+v", err)
+	}
+
+	return nil
+}
+
+func (a *mdsAgent) DestroyLocalService(context *clusterd.Context) error {
+	if a.mdsProc == nil {
+		logger.Debugf("no need to stop mds that is not running")
+		return nil
+	}
+
+	if err := a.mdsProc.Stop(); err != nil {
+		return fmt.Errorf("failed to stop mds. %+v", err)
+	}
+
+	logger.Infof("stopped ceph mds")
+	a.mdsProc = nil
+
+	// TODO: Clean up the mds folder
+	return nil
+}
+
+// create a keyring for the mds client with a limited set of privileges
+func createMDSKeyring(conn client.Connection, id, configDir string) error {
+	username := fmt.Sprintf("mds.%s", id)
+	keyringPath := getMDSKeyringPath(configDir, id)
+	access := []string{"osd", "allow *", "mds", "allow", "mon", "allow profile mds"}
+	keyringEval := func(key string) string {
+		return fmt.Sprintf(keyringTemplate, id, key)
+	}
+
+	return mon.CreateKeyring(conn, username, keyringPath, access, keyringEval)
+}
+
+func (a *mdsAgent) startMDS(context *clusterd.Context, cluster *mon.ClusterInfo, id string) error {
+
+	// start the monitor daemon in the foreground with the given config
+	logger.Infof("starting mds %s", id)
+	mdsNameArg := fmt.Sprintf("--name=mds.%s", id)
+	mdsProc, err := context.ProcMan.Start(
+		"mds",
+		regexp.QuoteMeta(mdsNameArg),
+		proc.ReuseExisting,
+		"--foreground",
+		mdsNameArg,
+		fmt.Sprintf("--cluster=%s", cluster.Name),
+		fmt.Sprintf("--conf=%s", getMDSConfFilePath(context.ConfigDir, id, cluster.Name)),
+		fmt.Sprintf("--keyring=%s", getMDSKeyringPath(context.ConfigDir, id)),
+		"-f", "-d", "-i", id)
```

Comment:
can we please use long form of all parameters to improve readability. is `-f` redundant?